### DISCUSSION TITLE
Form with 60 select inputs, watch problem

### DISCUSSION THREAD

**Discussion by Arsikod** [react-hook-form/react-hook-form#5400] [Q&A] [answered]
Hey guys,

So I have a form of 60 select input fields, and I dont write each component one by one. I generate them through mapping initial object that contains all the data related to component. I set there watchComponent relations as well to set watch relations.
Since I operate from single component I put watch constant inside this single component.

Problem is that every time i select other fields, all of there componets get triggered because of watch.

Question: 
1) Is there any chance to isolate watch depending on watcher and watching components? 

2) Is there any best practice recommended by RHF on generating big forms with more than 50-60 inputs? (in order not to write them one by one)

Here is a sandbox for simplified version, but the logic and architecture is the same
https://codesandbox.io/s/naughty-jackson-ghfng

**Comment by bluebill1049** [comment, marked as answer]
use `useWatch`. https://react-hook-form.com/api/usewatch

**Reply by Arsikod** [reply]
thank you sir for your prompt response, I really appreciate it.
Can you recommend any best practice with RHF on creating big forms (+60 inputs) ? (question 2)

**Reply by bluebill1049** [reply]
I don't think there is best practice on this one, because it really depends on your UI. If they r just repeat items, which needs to be manipulated, then I will useFieldArry, otherwise just map those inputs and register them.
